app/modules/code_provider/provider_factory.py

```
@@ -0,0 +1,368 @@
+            else None
+        )
+        is_github_app_configured = bool(app_id and private_key)
+
+        # For GitHub with App configured: Try GitHub App first, then PAT
+        if provider_type == ProviderType.GITHUB and is_github_app_configured:
+            logger.info(
+                f"GitHub App is configured, trying App auth first for {repo_name}"
+            )
+            try:
+                return CodeProviderFactory.create_github_app_provider(repo_name)
+            except Exception as e:
```

Ruff (0.14.2)

Do not catch blind exception: `Exception`

(BLE001)